vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -224,6 +224,14 @@ protected void fireSelectionEvent(
                             SelectionEvent<Grid<T>, T> event) {
                         grid.fireEvent((ComponentEvent<Grid<T>>) event);
                     }
+
+                    @Override
+                    public void setDeselectAllowed(boolean deselectAllowed) {
+                        super.setDeselectAllowed(deselectAllowed);
+                        grid.getElement().executeJs(
+                                "this.$connector.deselectAllowed = $0",
+                                deselectAllowed);
+                    }
```

Comment:
As the deselection is prevented only in the client-side, this is currently a pure UX feature, which can be hacked by running `grid.deselectAllowed = true` in the browser.

As a developer, I might expect that after calling `setDeselectAllowed(false)`, the value can't change to `null`.
If the developer is not cautious and handle the value change event with `null`, a hackerman can break the app.

The JavaDocs states "Sets whether it's allowed to deselect the selected row through UI", which kind-of indicates that it should be more of an UX-feature. Still, I think that it would be better to prevent deselection also at the server-side to avoid security issues.

Reply:
As discussed f2f, adding a check [here](https://github.com/vaadin/vaadin-grid-flow/blob/ca8cb26b9de724b0f4579067daceba94773800ab/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridSingleSelectionModel.java#L88) would probably be the best way to fix it so that would still allow deselection with the Java APIs

Reply:
Added check

Reply:
It should be actually without `!`. Currently the deselection is broken, as it doesn't update the server-side state to `null` value when deselection is allowed.

Reply:
True that. Fixed.